File: academy_Backend/services/employee_service.py
from models.employee import (
    get_all_employees,
    get_employee_by_id,
    get_employees_by_batch,
    create_employee,
    update_employee,
    delete_employee
)
from models.employee import (
    get_all_employees,
    get_employee_by_id,
    get_employees_by_batch,
    create_employee,
    update_employee,
    delete_employee,
    get_employee_count   # <-- add this import
)



def service_get_all_employees():
    try:
        records = get_all_employees()
        return {"success": True, "message": "Employees retrieved successfully", "data": records}
    except Exception as e:
        return {"success": False, "message": f"Error retrieving employees: {e}", "data": None}


def service_get_employee_by_id(employee_id: str):
    record = get_employee_by_id(employee_id)
    if record:
        return {"success": True, "message": "Employee retrieved successfully", "data": record}
    else:
        return {"success": False, "message": "Employee not found", "data": None}


def service_get_employees_by_batch(batch_code: int):
    try:
        records = get_employees_by_batch(batch_code)
        return {"success": True, "message": "Employees retrieved successfully", "data": records}
    except Exception as e:
        return {"success": False, "message": f"Error retrieving employees by batch: {e}", "data": None}


def service_create_employee(data):
    success = create_employee(
        data.user_id,
        data.pod_id,
        data.employee_name,
        data.employee_id,
        data.employee_email,
        data.designation
    )
    return {"success": success, "message": "Employee created successfully" if success else "Failed to create employee", "data": None}


def service_update_employee(employee_id, data):
    success = update_employee(
        employee_id,
        data.pod_id,
        data.employee_name,
        data.employee_email,
        data.designation
    )
    return {"success": success, "message": "Employee updated successfully" if success else "Failed to update employee", "data": None}


def service_delete_employee(employee_id: str):
    success = delete_employee(employee_id)
    return {"success": success, "message": "Employee deleted successfully" if success else "Failed to delete employee", "data": None}


def service_get_employee_count():
    try:
        count = get_employee_count()
        return {"success": True, "message": "Employee count retrieved successfully", "data": {"total_employees": count}}
    except Exception as e:
        return {"success": False, "message": f"Error retrieving employee count: {e}", "data": None}

File: academy_UI/src/app/shared/services/employee.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Employee {
  Employee_ID: string;
  Employee_Name: string;
  Employee_Email: string;
  Designation: string;
  POD_ID: number;
  POD: string;
  Batch_Code: number;
  assessment?: number;
  course_completion_percent ?: number;
  certification_completion_percent ?: number;
  assessment_completion_percent ?: number;
}

export interface EmployeeResponse {
  success: boolean;
  message: string;
  data: Employee[];
}

@Injectable({
  providedIn: 'root'
})
export class EmployeeService {
  private apiUrl = 'http://127.0.0.1:8000/api';

  constructor(private http: HttpClient) {}

  getEmployeesByBatch(batchCode: number): Observable<EmployeeResponse> {
    return this.http.get<EmployeeResponse>(`${this.apiUrl}/employees/batch/${batchCode}`);
  }

  getAllEmployees(): Observable<EmployeeResponse> {
    return this.http.get<EmployeeResponse>(`${this.apiUrl}/employees/get-all`);
  }

  getEmployeeById(employeeId: string): Observable<any> {
    return this.http.get<any>(`${this.apiUrl}/employees/${employeeId}`);
  }
}